django/db/models/functions/window.py

```diff
@@ -0,0 +1,120 @@
+from django.db.models import Aggregate, FloatField, IntegerField
+
+__all__ = [
+    'RowNumber', 'Rank', 'DenseRank', 'Ntile', 'FirstValue',
+    'PercentRank', 'NthValue', 'LastValue', 'CumeDist', 'Lag',
+    'Lead',
+]
+
+
+class CumeDist(Aggregate):
+    function = 'CUME_DIST'
+    name = 'CumeDist'
+
+    def __init__(self, *expressions, **extra):
+        super().__init__(*expressions, output_field=FloatField())
+
+
+class DenseRank(Aggregate):
+    function = 'DENSE_RANK'
+    name = 'DenseRank'
+
+    def __init__(self, *expressions, **extra):
+        super().__init__(*expressions, output_field=IntegerField())
+
+
+class FirstLastValueWrapper(Aggregate):
+    def __init__(self, expression, **extra):
+        if expression is None:
+            raise ValueError('%s requires a non-null source_expression' % self.__class__.__name__)
```

Comment:
I think this is the only place where `arity` makes sense. For the other functions, some of the arguments are optional which just raises an exception.